evm/generators.go

```diff
@@ -1066,6 +1072,229 @@ func Create{{.StructName}}Command() *cobra.Command {
+	}
+
+	safeTransactionData := SafeTransactionData{
+		To:             to.Hex(),
+		Value:          value.String(),
+		Data:           common.Bytes2Hex(data),
+		Operation:      safeOperation,
+		SafeTxGas:      0,
+		BaseGas:        0,
+		GasPrice:       "0",
+		GasToken:       NativeTokenAddress,
+		RefundReceiver: NativeTokenAddress, 
+		Nonce:          nonce.Uint64(),
+	}
+
+	// Calculate SafeTxHash
+	safeTxHash, err := CalculateSafeTxHash(safeAddress, safeTransactionData, chainID)
+	if err != nil {
+		return fmt.Errorf("failed to calculate SafeTxHash: %v", err)
+	}
+
+	// Sign the SafeTxHash
+	signature, err := crypto.Sign(safeTxHash.Bytes(), key.PrivateKey)
+	if err != nil {
+		return fmt.Errorf("failed to sign SafeTxHash: %v", err)
+	}
+
+	// Adjust V value for Ethereum's replay protection
+	signature[64] += 27
+
+	// Convert signature to hex
+	senderSignature := "0x" + common.Bytes2Hex(signature)
+
+	// Prepare the request body
+	requestBody := map[string]interface{}{
+		"to":             safeTransactionData.To,
+		"value":          safeTransactionData.Value,
+		"data":           "0x" + safeTransactionData.Data,
+		"operation":      int(safeTransactionData.Operation),
+		"safeTxGas":      fmt.Sprintf("%d", safeTransactionData.SafeTxGas),
+		"baseGas":        fmt.Sprintf("%d", safeTransactionData.BaseGas),
+		"gasPrice":       safeTransactionData.GasPrice,
+		"gasToken":       safeTransactionData.GasToken,
+		"refundReceiver": safeTransactionData.RefundReceiver,
+		"nonce":          fmt.Sprintf("%d", safeTransactionData.Nonce),
+		"safeTxHash":     safeTxHash.Hex(),
+		"sender":         key.Address.Hex(),
+		"signature":      senderSignature,
+		"origin":         fmt.Sprintf("{\"url\":\"%s\",\"name\":\"TokenSender Deployment\"}", txServiceBaseUrl),
+	}
+
+	// Marshal the request body to JSON
+	jsonBody, err := json.Marshal(requestBody)
+	if err != nil {
+		return fmt.Errorf("failed to marshal request body: %v", err)
+	}
+
+	// Send the request to the Safe Transaction Service
+	req, err := http.NewRequest("POST", txServiceBaseUrl, bytes.NewBuffer(jsonBody))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %v", err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	httpClient := &http.Client{}
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to send request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusCreated {
+		return fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	fmt.Println("Safe proposal created successfully")
+	return nil
+}
+
+func CalculateSafeTxHash(safeAddress common.Address, txData SafeTransactionData, chainID *big.Int) (common.Hash, error) {
+	domainSeparator := apitypes.TypedDataDomain{
+		ChainId:           (*math.HexOrDecimal256)(chainID),
+		VerifyingContract: safeAddress.Hex(),
+	}
+
+	typedData := apitypes.TypedData{
```

Comment:
Eventually, this should be an exported field from the [`G7DAO/safes`](https://github.com/G7DAO/safes) library. (see comment regarding this above, not a blocker for this PR).

Reply:
I will create an issue for that in both repositories

Reply:
Issues created: 

- https://github.com/moonstream-to/seer/issues/74
- https://github.com/G7DAO/safes/issues/1